tests/auth.spec.js

```diff
@@ -0,0 +1,81 @@
+import { describe } from 'mocha';
+import chai, { expect } from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../index';
+import { User } from '../server/models';
+
+chai.use(chaiHttp);
+
+describe('Test Cases for the Authentication end points', () => {
+  describe('Test Cases for Login Endpoint', () => {
+    let userData;
+    before((done) => {
+      userData = {
+        email: 'user1@example.test',
+        password: 'secretpassword',
+        firstname: 'Johnny',
+        lastname: 'Bravo',
+        username: 'johnnybravo'
+      };
+      chai.request(app)
+        .post('/api/users')
+        .send(userData)
+        .end(() => {
+          done();
+        });
+    });
+    after(async () => {
+      // delete user
+      User.destroy({ where: { email: userData.email } });
```

Comment:
@jsamchineme Consider recreating all tables in the db after running the test suite, rather than just deleting records. Something like this:  ``` after((done) => { //Before each test we empty the database
    models.sequelize.sync({
      force: true
    })
      .then(() => {
        done();
      });
  }); ``` or  You can systematically drop the parent models first before their dependants like so
 
 ```  after((done) => {
    db.Documents.destroy({where: {}}).then(() => {
      db.Users.destroy({ where: {} }).then(() => {
        db.Roles.destroy({ where: {} }).then(()  => {
          done();
        });
      })
    })
    .catch(error => console.log(error));
  })

```